WordPress/gutenberg
# Docs: Unify indentation styles in markdown files

## What?
This PR unifies tab indentation for markdown files in the `docs` directory.

## Why?
In many files, space indentation and tab indentation are mixed. In addition, the number of spaces in a space indent varies. As indicated by [.editorconfig](https://github.com/WordPress/gutenberg/blob/3d85dcb89f6a1081a184560c628ddd49396407c3/.editorconfig#L9-L14), tab indentation should be uniform in markdown files.

## How?
I tried to detect incorrect indentation using the following conditions:

- Lines begin with **two spaces**
- Target markdown files in the `docs` directory
- However, exclude automatically generated markdown files in `reference-guides/data`

From the results obtained from this filter, I further excluded space indentation caused by list tags.

In addition to unifying indentation, spaces at the end of lines are also removed (this is due to the code editor's saving behavior).

## Testing Instructions
- See diff (If you view it in [split mode](https://github.com/WordPress/gutenberg/pull/44793/files?diff=split&w=0), it is easy to see what this PR does).
- To confirm that no changes have been made to the auto-generated documentation, run `docs:build` to confirm that no differences occur.
